src/components/UserProfile.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Button,
+  Card,
+  CardContent,
+  CardActions,
+  Table,
+  TableRow,
+  TableCell,
+  getImageListItemBarUtilityClass,
+} from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+// import { getUserProfile } from './api';
+import { Avatar } from '@mui/material';
+import axios from 'axios';
+
+export default function PersonalProfile() {
+  const navigate = useNavigate();
+  const { user, status } = useAuth();
+  const [userData, setUserData] = useState(null); 
```

Comment:
Instead of setting the state from useEffect, we can just use the user that was returned from the `useAuth` hook in the previous line:

```suggestion
  const userData = authUser.user;
```